### PR TITLE
Enterprise3.4

### PR DESCRIPTION
**Pull request by nicolaskruchten**

### CODE REVIEW

File: src/default_panels/StyleLegendPanel.js [hunk 1]
```diff
@@ -21,6 +22,13 @@ const StyleLegendPanel = (props, {localize: _}) => (
           {label: _('Hide'), value: false},
         ]}
       />
+
+      <PlotlySection name={_('Legend Title')}>
+        <TextEditor label={_('text')} attr="legend.title.text" richTextOnly />
```
Comment (dmt0) [inline src/default_panels/StyleLegendPanel.js]:
```suggestion
        <TextEditor label={_('Text')} attr="legend.title.text" richTextOnly />
```

File: package.json [hunk 2]
```diff
@@ -35,7 +35,7 @@
     "fast-isnumeric": "^1.1.2",
     "immutability-helper": "^3.0.0",
     "plotly-icons": "1.3.13",
-    "plotly.js": "1.51.3",
+    "plotly.js": "1.52.0",
```
Comment (dmt0) [inline package.json]:
```suggestion
    "plotly.js": "1.52.1",
```

File: src/default_panels/StyleTracesPanel.js [hunk 3]
```diff
@@ -653,11 +659,38 @@ const StyleTracesPanel = (props, {localize: _}) => (
       )}
       mode="trace"
     >
-      <DataSelector label={_('Text')} attr="text" />
       <TextPosition label={_('Text Position')} attr="textposition" />
+      <HoverTemplateSwitch attr="texttemplate" label={_('Mode')} />
+      <TextInfo attr="textinfo" label={_('Show')} />
+      <HoverTemplateText attr="texttemplate" label={_('Template')} />
```
Comment (dmt0) [inline src/default_panels/StyleTracesPanel.js]: `attr="texttemplate"` - we have it twice here?

File: package.json [hunk 4]
```diff
@@ -35,7 +35,7 @@
     "fast-isnumeric": "^1.1.2",
     "immutability-helper": "^3.0.0",
     "plotly-icons": "1.3.13",
```
Comment (dmt0) [inline package.json]:
```suggestion
    "plotly-icons": "1.3.14",
```

File: src/default_panels/StyleTracesPanel.js (hunk 3 above)
Comment (nicolaskruchten) [reply src/default_panels/StyleTracesPanel.js]: it's a too-clever way of building the "switch" control

Comment (nicolaskruchten) [reply src/default_panels/StyleTracesPanel.js]: one is the switch and the other is the template text

**Comment by nicolaskruchten** [comment]
Thanks for the review!

**Pull request merged**